### ISSUE TITLE
The GPU utilization rate is low during training, and the training is very slow！

### ISSUE THREAD

**Issue by tms2003** [Peterande/D-FINE#5]
I use the following command to train on a GPU card. The GPU is generally less than 10%. Why is the training so low?

 torchrun  train.py -c configs/dfine/custom/dfine_hgnetv2_l_custom.yml --use-amp --seed=0
This's my config.xml
```
task: detection

evaluator:
  type: CocoEvaluator
  iou_types: ['bbox', ]

num_classes: 11 # your dataset classes
remap_mscoco_category: False

train_dataloader: 
  type: DataLoader
  dataset: 
    type: CocoDetection
    img_folder: /tms/data/coco_2024/images
    ann_file: /tms/data/coco_2024/annotations/result.json
    return_masks: False
    transforms:
      type: Compose
      ops: ~
  shuffle: True
  num_workers: 8
  drop_last: True 
  collate_fn:
    type: BatchImageCollateFuncion


val_dataloader:
  type: DataLoader
  dataset: 
    type: CocoDetection
    img_folder: /tms/data/coco_2024/images
    ann_file: /tms/data/coco_2024/annotations/val.json
    return_masks: False
    transforms:
      type: Compose
      ops: ~ 
  shuffle: False
  num_workers: 8
  drop_last: False
  collate_fn:
    type: BatchImageCollateFuncion

```
This is the first output:

```
Epoch: [0]  [   0/3999]  eta: 8:34:10  lr: 0.000000  loss: 64.9147 (64.9147)  loss_vfl: 0.2810 (0.2810)  loss_bbox: 2.2343 (2.2343)  loss_giou: 1.9365 (1.9365)  loss_fgl: 0.2977 (0.2977)  loss_vfl_aux_0: 0.2856 (0.2856)  loss_bbox_aux_0: 2.2343 (2.2343)  loss_giou_aux_0: 1.9365 (1.9365)  loss_fgl_aux_0: 0.2977 (0.2977)  loss_vfl_aux_1: 0.2942 (0.2942)  loss_bbox_aux_1: 2.2343 (2.2343)  loss_giou_aux_1: 1.9365 (1.9365)  loss_fgl_aux_1: 0.2977 (0.2977)  loss_vfl_aux_2: 0.2607 (0.2607)  loss_bbox_aux_2: 2.2343 (2.2343)  loss_giou_aux_2: 1.9365 (1.9365)  loss_fgl_aux_2: 0.2977 (0.2977)  loss_vfl_aux_3: 0.2947 (0.2947)  loss_bbox_aux_3: 2.2343 (2.2343)  loss_giou_aux_3: 1.9365 (1.9365)  loss_fgl_aux_3: 0.2977 (0.2977)  loss_vfl_aux_4: 0.2991 (0.2991)  loss_bbox_aux_4: 2.2343 (2.2343)  loss_giou_aux_4: 1.9365 (1.9365)  loss_fgl_aux_4: 0.2977 (0.2977)  loss_vfl_pre: 0.2856 (0.2856)  loss_bbox_pre: 2.2343 (2.2343)  loss_giou_pre: 1.9365 (1.9365)  loss_vfl_enc_0: 0.2507 (0.2507)  loss_bbox_enc_0: 2.2343 (2.2343)  loss_giou_enc_0: 1.9365 (1.9365)  loss_vfl_dn_0: 0.8105 (0.8105)  loss_bbox_dn_0: 1.0030 (1.0030)  loss_giou_dn_0: 1.3419 (1.3419)  loss_fgl_dn_0: 0.8430 (0.8430)  loss_vfl_dn_1: 0.8896 (0.8896)  loss_bbox_dn_1: 1.0030 (1.0030)  loss_giou_dn_1: 1.3419 (1.3419)  loss_fgl_dn_1: 0.8430 (0.8430)  loss_vfl_dn_2: 0.9038 (0.9038)  loss_bbox_dn_2: 1.0030 (1.0030)  loss_giou_dn_2: 1.3419 (1.3419)  loss_fgl_dn_2: 0.8430 (0.8430)  loss_vfl_dn_3: 0.8555 (0.8555)  loss_bbox_dn_3: 1.0030 (1.0030)  loss_giou_dn_3: 1.3419 (1.3419)  loss_fgl_dn_3: 0.8430 (0.8430)  loss_vfl_dn_4: 0.8955 (0.8955)  loss_bbox_dn_4: 1.0030 (1.0030)  loss_giou_dn_4: 1.3419 (1.3419)  loss_fgl_dn_4: 0.8430 (0.8430)  loss_vfl_dn_5: 0.8726 (0.8726)  loss_bbox_dn_5: 1.0030 (1.0030)  loss_giou_dn_5: 1.3419 (1.3419)  loss_fgl_dn_5: 0.8430 (0.8430)  loss_vfl_dn_pre: 0.8105 (0.8105)  loss_bbox_dn_pre: 1.0030 (1.0030)  loss_giou_dn_pre: 1.3419 (1.3419)  time: 7.7145  data: 3.3224  max mem: 15826
Epoch: [0]  [ 100/3999]  eta: 1:47:07  lr: 0.000003  loss: 54.7667 (58.6890)  loss_vfl: 0.5742 (0.4503)  loss_bbox: 1.3285 (1.7103)  loss_giou: 1.4214 (1.6404)  loss_fgl: 0.6612 (0.5103)  loss_vfl_aux_0: 0.4985 (0.3966)  loss_bbox_aux_0: 1.3807 (1.7359)  loss_giou_aux_0: 1.4401 (1.6587)  loss_fgl_aux_0: 0.6462 (0.5032)  loss_vfl_aux_1: 0.5386 (0.4283)  loss_bbox_aux_1: 1.3580 (1.7269)  loss_giou_aux_1: 1.4328 (1.6523)  loss_fgl_aux_1: 0.6526 (0.5057)  loss_vfl_aux_2: 0.5366 (0.4177)  loss_bbox_aux_2: 1.3494 (1.7219)  loss_giou_aux_2: 1.4304 (1.6486)  loss_fgl_aux_2: 0.6551 (0.5069)  loss_vfl_aux_3: 0.5562 (0.4359)  loss_bbox_aux_3: 1.3402 (1.7171)  loss_giou_aux_3: 1.4273 (1.6454)  loss_fgl_aux_3: 0.6579 (0.5082)  loss_vfl_aux_4: 0.5854 (0.4475)  loss_bbox_aux_4: 1.3354 (1.7137)  loss_giou_aux_4: 1.4249 (1.6427)  loss_fgl_aux_4: 0.6596 (0.5093)  loss_vfl_pre: 0.4954 (0.3937)  loss_bbox_pre: 1.4076 (1.7411)  loss_giou_pre: 1.4508 (1.6630)  loss_vfl_enc_0: 0.4766 (0.3814)  loss_bbox_enc_0: 1.4723 (1.7543)  loss_giou_enc_0: 1.4959 (1.6754)  loss_vfl_dn_0: 0.4158 (0.5894)  loss_bbox_dn_0: 0.9358 (0.9461)  loss_giou_dn_0: 1.3314 (1.3310)  loss_fgl_dn_0: 0.8526 (0.8504)  loss_vfl_dn_1: 0.4136 (0.6395)  loss_bbox_dn_1: 0.9422 (0.9496)  loss_giou_dn_1: 1.3315 (1.3316)  loss_fgl_dn_1: 0.8487 (0.8491)  loss_vfl_dn_2: 0.4224 (0.5959)  loss_bbox_dn_2: 0.9394 (0.9516)  loss_giou_dn_2: 1.3293 (1.3318)  loss_fgl_dn_2: 0.8433 (0.8472)  loss_vfl_dn_3: 0.4158 (0.5715)  loss_bbox_dn_3: 0.9378 (0.9543)  loss_giou_dn_3: 1.3285 (1.3322)  loss_fgl_dn_3: 0.8406 (0.8459)  loss_vfl_dn_4: 0.4233 (0.6076)  loss_bbox_dn_4: 0.9366 (0.9570)  loss_giou_dn_4: 1.3281 (1.3324)  loss_fgl_dn_4: 0.8386 (0.8448)  loss_vfl_dn_5: 0.4246 (0.5783)  loss_bbox_dn_5: 0.9348 (0.9590)  loss_giou_dn_5: 1.3261 (1.3325)  loss_fgl_dn_5: 0.8368 (0.8437)  loss_vfl_dn_pre: 0.4163 (0.5897)  loss_bbox_dn_pre: 0.9320 (0.9444)  loss_giou_dn_pre: 1.3310 (1.3308)  loss_ddf_aux_0: 0.0068 (0.0026)  loss_ddf_aux_1: 0.0042 (0.0015)  loss_ddf_aux_2: 0.0020 (0.0008)  loss_ddf_aux_3: 0.0010 (0.0003)  loss_ddf_aux_4: 0.0003 (0.0001)  loss_ddf_dn_0: 0.0043 (0.0017)  loss_ddf_dn_1: 0.0031 (0.0012)  loss_ddf_dn_2: 0.0016 (0.0006)  loss_ddf_dn_3: 0.0007 (0.0003)  loss_ddf_dn_4: 0.0002 (0.0001)  time: 1.5149  data: 0.0414  max mem: 33922

```
In addition, if you do not use torchrun, you will get an error when running train.py directly with python.

**Comment by Peterande** [comment]
According to the log provided in other people's issues, the training speed should be normal. We have also tested on no less than dozens of servers. May I ask what is your current total batch size? On what kind of single GPU are you training? You can try setting num_workers to 0 to see if the problem can be solved.

**Comment by Peterande** [comment]
I will check the problem about "running train.py directly" later.

**Comment by Peterande** [comment]
> According to the log provided in other people's issues, the training speed should be normal. We have also tested on no less than dozens of servers. May I ask what is your current total batch size? On what kind of single GPU are you training? You can try setting num_workers to 0 to see if the problem can be solved.

You need to modify nun_workers here: https://github.com/Peterande/D-FINE/blob/b27b1d8e596b83d98604958eba96761c027e7f54/configs/dfine/include/dataloader.yml#L28

**Comment by tms2003** [comment]
I already changed num_workers: 0 in dataloader.yml but Reserved myself's _detection.yml with num_workers: 32 but the result is still that slow.
my log like this:
```
torchrun  train.py -c configs/dfine/custom/dfine_hgnetv2_l_custom.yml --use-amp --seed=0                            
Initialized distributed mode...                                                                                                                                          
cfg:  {'task': 'detection', '_model': None, '_postprocessor': None, '_criterion': None, '_optimizer': None, '_lr_scheduler': None, '_lr_warmup_scheduler': None, '_train_
dataloader': None, '_val_dataloader': None, '_ema': None, '_scaler': None, '_train_dataset': None, '_val_dataset': None, '_collate_fn': None, '_evaluator': None, '_write
r': None, 'num_workers': 0, 'batch_size': None, '_train_batch_size': None, '_val_batch_size': None, '_train_shuffle': None, '_val_shuffle': None, 'resume': None, 'tuning
': None, 'epoches': 80, 'last_epoch': -1, 'use_amp': True, 'use_ema': True, 'ema_decay': 0.9999, 'ema_warmups': 2000, 'sync_bn': True, 'clip_max_norm': 0.1, 'find_unused
_parameters': False, 'seed': 0, 'print_freq': 100, 'checkpoint_freq': 12, 'output_dir': './output/dfine_hgnetv2_l_custom', 'summary_dir': None, 'device': '', 'yaml_cfg':
 {'task': 'detection', 'evaluator': {'type': 'CocoEvaluator', 'iou_types': ['bbox']}, 'num_classes': 11, 'remap_mscoco_category': False, 'train_dataloader': {'type': 'Da
taLoader', 'dataset': {'type': 'CocoDetection', 'img_folder': '/tms/data/coco_2024/images', 'ann_file': '/tms/data/coco_2024/annotations/result.json', 'return_ma
sks': False, 'transforms': {'type': 'Compose', 'ops': [{'type': 'RandomPhotometricDistort', 'p': 0.5}, {'type': 'RandomZoomOut', 'fill': 0}, {'type': 'RandomIoUCrop', 'p
': 0.8}, {'type': 'SanitizeBoundingBoxes', 'min_size': 1}, {'type': 'RandomHorizontalFlip'}, {'type': 'Resize', 'size': [640, 640]}, {'type': 'SanitizeBoundingBoxes', 'm
in_size': 1}, {'type': 'ConvertPILImage', 'dtype': 'float32', 'scale': True}, {'type': 'ConvertBoxes', 'fmt': 'cxcywh', 'normalize': True}], 'policy': {'name': 'stop_epo
ch', 'epoch': 72, 'ops': ['RandomPhotometricDistort', 'RandomZoomOut', 'RandomIoUCrop']}}}, 'shuffle': True, 'num_workers': 0, 'drop_last': True, 'collate_fn': {'type': 
'BatchImageCollateFuncion', 'scales': [480, 512, 544, 576, 608, 672, 704, 736, 768, 800], 'scale_ori_repeat': 4, 'stop_epoch': 72, 'ema_restart_decay': 0.9999}, 'total_b
atch_size': 32}, 'val_dataloader': {'type': 'DataLoader', 'dataset': {'type': 'CocoDetection', 'img_folder': '/tms/data/coco_2024/images', 'ann_file': '/tms/data
/coco_2024/annotations/val.json', 'return_masks': False, 'transforms': {'type': 'Compose', 'ops': [{'type': 'Resize', 'size': [640, 640]}, {'type': 'ConvertPILImage', 'd
type': 'float32', 'scale': True}]}}, 'shuffle': False, 'num_workers': 0, 'drop_last': False, 'collate_fn': {'type': 'BatchImageCollateFuncion'}, 'total_batch_size': 64},
 'print_freq': 100, 'output_dir': './output/dfine_hgnetv2_l_custom', 'checkpoint_freq': 12, 'sync_bn': True, 'find_unused_parameters': False, 'use_amp': True, 'scaler': 
{'type': 'GradScaler', 'enabled': True}, 'use_ema': True, 'ema': {'type': 'ModelEMA', 'decay': 0.9999, 'warmups': 1000, 'start': 0}, 'epoches': 80, 'clip_max_norm': 0.1,
 'optimizer': {'type': 'AdamW', 'params': [{'params': '^(?=.*backbone)(?!.*norm|bn).*$', 'lr': 1.25e-05}, {'params': '^(?=.*(?:encoder|decoder))(?=.*(?:norm|bn)).*$', 'w
eight_decay': 0.0}], 'lr': 0.00025, 'betas': [0.9, 0.999], 'weight_decay': 0.000125}, 'lr_scheduler': {'type': 'MultiStepLR', 'milestones': [500], 'gamma': 0.1}, 'lr_war
mup_scheduler': {'type': 'LinearWarmup', 'warmup_duration': 500}, 'model': 'DFINE', 'criterion': 'DFINECriterion', 'postprocessor': 'DFINEPostProcessor', 'use_focal_loss
': True, 'eval_spatial_size': [640, 640], 'DFINE': {'backbone': 'HGNetv2', 'encoder': 'HybridEncoder', 'decoder': 'DFINETransformer'}, 'HGNetv2': {'pretrained': True, 'l
ocal_model_dir': 'weight/hgnetv2/', 'name': 'B4', 'return_idx': [1, 2, 3], 'freeze_stem_only': True, 'freeze_at': 0, 'freeze_norm': True}, 'HybridEncoder': {'in_channels
': [512, 1024, 2048], 'feat_strides': [8, 16, 32], 'hidden_dim': 256, 'use_encoder_idx': [2], 'num_encoder_layers': 1, 'nhead': 8, 'dim_feedforward': 1024, 'dropout': 0.
0, 'enc_act': 'gelu', 'expansion': 1.0, 'depth_mult': 1, 'act': 'silu'}, 'DFINETransformer': {'feat_channels': [256, 256, 256], 'feat_strides': [8, 16, 32], 'hidden_dim': 256, 'num_levels': 3, 'num_layers': 6, 'eval_idx': -1, 'num_queries': 300, 'num_denoising': 100, 'label_noise_ratio': 0.5, 'box_noise_scale': 1.0, 'reg_max': 32, 'reg_scale': 4, 'layer_scale': 1, 'num_points': [3, 6, 3], 'cross_attn_method': 'default', 'query_select_method': 'default'}, 'DFINEPostProcessor': {'num_top_queries': 300}, 'DFINECriterion': {'weight_dict': {'loss_vfl': 1, 'loss_bbox': 5, 'loss_giou': 2, 'loss_fgl': 0.15, 'loss_ddf': 1.5}, 'losses': ['vfl', 'boxes', 'local'], 'alpha': 0.75, 'gamma': 2.0, 'reg_max': 32, 'matcher': {'type': 'HungarianMatcher', 'weight_dict': {'cost_class': 2, 'cost_bbox': 5, 'cost_giou': 2}, 'alpha': 0.25, 'gamma': 2.0}}, '__include__': ['../../dataset/c11_detection.yml', '../../runtime.yml', '../include/dataloader.yml', '../include/optimizer.yml', '../include/dfine_hgnetv2.yml'], 'config': 'configs/dfine/custom/dfine_hgnetv2_l_custom.yml', 'seed': 0, 'test_only': False, 'print_method': 'builtin', 'print_rank': 0}}
/tms/source/D-FINE/src/nn/backbone/hgnetv2.py:494: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  state = torch.load(model_path, map_location='cpu')
Loaded stage1 B4 HGNetV2 from local file.
/tms/source/D-FINE/src/core/workspace.py:181: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  return module(**module_kwargs)
Initial lr: [1.25e-05, 0.00025, 0.00025]
building train_dataloader with batch_size=32...
loading annotations into memory...
Done (t=3.90s)
creating index...
index created!
building val_dataloader with batch_size=64...
loading annotations into memory...
Done (t=0.38s)
creating index...
index created!

------------------------------------- Calculate Flops Results -------------------------------------
Notations:
number of parameters (Params), number of multiply-accumulate operations(MACs),
number of floating-point operations (FLOPs), floating-point operations per second (FLOPS),
fwd FLOPs (model forward propagation FLOPs), bwd FLOPs (model backward propagation FLOPs),
default model backpropagation takes 2.00 times as much computation as forward propagation.

Total Training Params:                                                  30.65 M 
fwd MACs:                                                               45.2972 GMACs
fwd FLOPs:                                                              90.7271 GFLOPS
fwd+bwd MACs:                                                           135.892 GMACs
fwd+bwd FLOPs:                                                          272.181 GFLOPS
---------------------------------------------------------------------------------------------------
{'Model FLOPs:90.7271 GFLOPS   MACs:45.2972 GMACs   Params:30671065'}
------------------------------------------Start training-------------------------------------------
Epoch: [0]  [   0/3999]  eta: 10:49:19  lr: 0.000000  loss: 66.6553 (66.6553)  loss_vfl: 0.2296 (0.2296)  loss_bbox: 2.3268 (2.3268)  loss_giou: 1.9382 (1.9382)  loss_fgl: 0.2838 (0.2838)  loss_vfl_aux_0: 0.2094 (0.2094)  loss_bbox_aux_0: 2.3268 (2.3268)  loss_giou_aux_0: 1.9382 (1.9382)  loss_fgl_aux_0: 0.2838 (0.2838)  loss_vfl_aux_1: 0.2266 (0.2266)  loss_bbox_aux_1: 2.3268 (2.3268)  loss_giou_aux_1: 1.9382 (1.9382)  loss_fgl_aux_1: 0.2838 (0.2838)  loss_vfl_aux_2: 0.2228 (0.2228)  loss_bbox_aux_2: 2.3268 (2.3268)  loss_giou_aux_2: 1.9382 (1.9382)  loss_fgl_aux_2: 0.2838 (0.2838)  loss_vfl_aux_3: 0.2313 (0.2313)  loss_bbox_aux_3: 2.3268 (2.3268)  loss_giou_aux_3: 1.9382 (1.9382)  loss_fgl_aux_3: 0.2838 (0.2838)  loss_vfl_aux_4: 0.2340 (0.2340)  loss_bbox_aux_4: 2.3268 (2.3268)  loss_giou_aux_4: 1.9382 (1.9382)  loss_fgl_aux_4: 0.2838 (0.2838)  loss_vfl_pre: 0.2094 (0.2094)  loss_bbox_pre: 2.3268 (2.3268)  loss_giou_pre: 1.9382 (1.9382)  loss_vfl_enc_0: 0.2062 (0.2062)  loss_bbox_enc_0: 2.3268 (2.3268)  loss_giou_enc_0: 1.9382 (1.9382)  loss_vfl_dn_0: 0.8623 (0.8623)  loss_bbox_dn_0: 1.1900 (1.1900)  loss_giou_dn_0: 1.3123 (1.3123)  loss_fgl_dn_0: 0.8643 (0.8643)  loss_vfl_dn_1: 0.9287 (0.9287)  loss_bbox_dn_1: 1.1900 (1.1900)  loss_giou_dn_1: 1.3123 (1.3123)  loss_fgl_dn_1: 0.8643 (0.8643)  loss_vfl_dn_2: 0.9683 (0.9683)  loss_bbox_dn_2: 1.1900 (1.1900)  loss_giou_dn_2: 1.3123 (1.3123)  loss_fgl_dn_2: 0.8643 (0.8643)  loss_vfl_dn_3: 0.8862 (0.8862)  loss_bbox_dn_3: 1.1900 (1.1900)  loss_giou_dn_3: 1.3123 (1.3123)  loss_fgl_dn_3: 0.8643 (0.8643)  loss_vfl_dn_4: 0.9385 (0.9385)  loss_bbox_dn_4: 1.1900 (1.1900)  loss_giou_dn_4: 1.3123 (1.3123)  loss_fgl_dn_4: 0.8643 (0.8643)  loss_vfl_dn_5: 0.9150 (0.9150)  loss_bbox_dn_5: 1.1900 (1.1900)  loss_giou_dn_5: 1.3123 (1.3123)  loss_fgl_dn_5: 0.8643 (0.8643)  loss_vfl_dn_pre: 0.8623 (0.8623)  loss_bbox_dn_pre: 1.1900 (1.1900)  loss_giou_dn_pre: 1.3123 (1.3123)  time: 9.7423  data: 7.2448  max mem: 23560
```
I run it on the only an  A100 GPU.

**Comment by Peterande** [comment]
I will try to reproduce this situation on a single GPU. However, regarding your output:

1. The ETA at the beginning of each epoch is usually inaccurate and will decrease rapidly within the first few hundred iterations.
2. Please use `nvitop` to check if any other processes are occupying the GPU resources.

**Comment by Peterande** [comment]
Does it still run slowly after you comment out all the TensorBoard operations?

https://github.com/Peterande/D-FINE/blob/b27b1d8e596b83d98604958eba96761c027e7f54/src/solver/det_engine.py#L107-L112

**Comment by Peterande** [comment]
RTX3090 reproduce, total batch size 16:
`
torchrun train.py -c configs/dfine/dfine_hgnetv2_${model}_coco.yml --use-amp --seed=0
`
![image](https://github.com/user-attachments/assets/d93e4fb0-12c2-41eb-b29b-b78c13f6418d)

```

Epoch: [0]  [   0/7392]  eta: 4:33:43  lr: 0.000000  loss: 59.3918 (59.3918)  loss_vfl: 0.3718 (0.3718)  loss_bbox: 1.5937 (1.5937)  loss_giou: 1.8046 (1.8046)  loss_fgl: 0.4020 (0.4020)  loss_vfl_aux_0: 0.4233 (0.4233)  loss_bbox_aux_0: 1.5937 (1.5937)  loss_giou_aux_0: 1.8046 (1.8046)  loss_fgl_aux_0: 0.4020 (0.4020)  loss_vfl_aux_1: 0.3875 (0.3875)  loss_bbox_aux_1: 1.5937 (1.5937)  loss_giou_aux_1: 1.8046 (1.8046)  loss_fgl_aux_1: 0.4020 (0.4020)  loss_vfl_aux_2: 0.3889 (0.3889)  loss_bbox_aux_2: 1.5937 (1.5937)  loss_giou_aux_2: 1.8046 (1.8046)  loss_fgl_aux_2: 0.4020 (0.4020)  loss_vfl_aux_3: 0.3877 (0.3877)  loss_bbox_aux_3: 1.5937 (1.5937)  loss_giou_aux_3: 1.8046 (1.8046)  loss_fgl_aux_3: 0.4020 (0.4020)  loss_vfl_aux_4: 0.3689 (0.3689)  loss_bbox_aux_4: 1.5937 (1.5937)  loss_giou_aux_4: 1.8046 (1.8046)  loss_fgl_aux_4: 0.4020 (0.4020)  loss_vfl_pre: 0.4233 (0.4233)  loss_bbox_pre: 1.5937 (1.5937)  loss_giou_pre: 1.8046 (1.8046)  loss_vfl_enc_0: 0.3911 (0.3911)  loss_bbox_enc_0: 1.5937 (1.5937)  loss_giou_enc_0: 1.8046 (1.8046)  loss_vfl_dn_0: 0.8760 (0.8760)  loss_bbox_dn_0: 0.8923 (0.8923)  loss_giou_dn_0: 1.3452 (1.3452)  loss_fgl_dn_0: 0.8456 (0.8456)  loss_vfl_dn_1: 0.8779 (0.8779)  loss_bbox_dn_1: 0.8923 (0.8923)  loss_giou_dn_1: 1.3452 (1.3452)  loss_fgl_dn_1: 0.8456 (0.8456)  loss_vfl_dn_2: 0.8154 (0.8154)  loss_bbox_dn_2: 0.8923 (0.8923)  loss_giou_dn_2: 1.3452 (1.3452)  loss_fgl_dn_2: 0.8456 (0.8456)  loss_vfl_dn_3: 0.8086 (0.8086)  loss_bbox_dn_3: 0.8923 (0.8923)  loss_giou_dn_3: 1.3452 (1.3452)  loss_fgl_dn_3: 0.8456 (0.8456)  loss_vfl_dn_4: 0.8193 (0.8193)  loss_bbox_dn_4: 0.8923 (0.8923)  loss_giou_dn_4: 1.3452 (1.3452)  loss_fgl_dn_4: 0.8456 (0.8456)  loss_vfl_dn_5: 0.8418 (0.8418)  loss_bbox_dn_5: 0.8923 (0.8923)  loss_giou_dn_5: 1.3452 (1.3452)  loss_fgl_dn_5: 0.8456 (0.8456)  loss_vfl_dn_pre: 0.8760 (0.8760)  loss_bbox_dn_pre: 0.8923 (0.8923)  loss_giou_dn_pre: 1.3452 (1.3452)  time: 2.2217  data: 1.2393  max mem: 15112
Epoch: [0]  [ 100/7392]  eta: 1:13:07  lr: 0.000003  loss: 56.2025 (57.9531)  loss_vfl: 0.4348 (0.3721)  loss_bbox: 1.3989 (1.5938)  loss_giou: 1.7223 (1.8009)  loss_fgl: 0.4563 (0.4028)  loss_vfl_aux_0: 0.3728 (0.3391)  loss_bbox_aux_0: 1.4197 (1.6038)  loss_giou_aux_0: 1.7470 (1.8163)  loss_fgl_aux_0: 0.4463 (0.3969)  loss_vfl_aux_1: 0.4009 (0.3478)  loss_bbox_aux_1: 1.4084 (1.6003)  loss_giou_aux_1: 1.7344 (1.8108)  loss_fgl_aux_1: 0.4511 (0.3991)  loss_vfl_aux_2: 0.4009 (0.3517)  loss_bbox_aux_2: 1.4039 (1.5982)  loss_giou_aux_2: 1.7284 (1.8076)  loss_fgl_aux_2: 0.4523 (0.4003)  loss_vfl_aux_3: 0.4331 (0.3593)  loss_bbox_aux_3: 1.4011 (1.5964)  loss_giou_aux_3: 1.7252 (1.8048)  loss_fgl_aux_3: 0.4542 (0.4013)  loss_vfl_aux_4: 0.4377 (0.3639)  loss_bbox_aux_4: 1.3996 (1.5948)  loss_giou_aux_4: 1.7229 (1.8024)  loss_fgl_aux_4: 0.4557 (0.4022)  loss_vfl_pre: 0.3682 (0.3372)  loss_bbox_pre: 1.4252 (1.6057)  loss_giou_pre: 1.7543 (1.8191)  loss_vfl_enc_0: 0.3525 (0.3376)  loss_bbox_enc_0: 1.4507 (1.6135)  loss_giou_enc_0: 1.7840 (1.8316)  loss_vfl_dn_0: 0.6104 (0.7405)  loss_bbox_dn_0: 0.8605 (0.8423)  loss_giou_dn_0: 1.3456 (1.3426)  loss_fgl_dn_0: 0.8352 (0.8420)  loss_vfl_dn_1: 0.5996 (0.7327)  loss_bbox_dn_1: 0.8679 (0.8460)  loss_giou_dn_1: 1.3464 (1.3430)  loss_fgl_dn_1: 0.8299 (0.8406)  loss_vfl_dn_2: 0.5942 (0.7057)  loss_bbox_dn_2: 0.8700 (0.8487)  loss_giou_dn_2: 1.3474 (1.3434)  loss_fgl_dn_2: 0.8264 (0.8391)  loss_vfl_dn_3: 0.5928 (0.7130)  loss_bbox_dn_3: 0.8708 (0.8515)  loss_giou_dn_3: 1.3481 (1.3438)  loss_fgl_dn_3: 0.8233 (0.8377)  loss_vfl_dn_4: 0.6069 (0.7118)  loss_bbox_dn_4: 0.8698 (0.8544)  loss_giou_dn_4: 1.3489 (1.3442)  loss_fgl_dn_4: 0.8201 (0.8365)  loss_vfl_dn_5: 0.5972 (0.7139)  loss_bbox_dn_5: 0.8672 (0.8562)  loss_giou_dn_5: 1.3500 (1.3445)  loss_fgl_dn_5: 0.8192 (0.8356)  loss_vfl_dn_pre: 0.6128 (0.7413)  loss_bbox_dn_pre: 0.8571 (0.8407)  loss_giou_dn_pre: 1.3453 (1.3424)  loss_ddf_aux_0: 0.0046 (0.0020)  loss_ddf_aux_1: 0.0027 (0.0012)  loss_ddf_aux_2: 0.0015 (0.0007)  loss_ddf_aux_3: 0.0006 (0.0003)  loss_ddf_aux_4: 0.0002 (0.0001)  loss_ddf_dn_0: 0.0057 (0.0023)  loss_ddf_dn_1: 0.0035 (0.0016)  loss_ddf_dn_2: 0.0019 (0.0009)  loss_ddf_dn_3: 0.0008 (0.0004)  loss_ddf_dn_4: 0.0002 (0.0001)  time: 0.5679  data: 0.0320  max mem: 17561
```


Everything seems to be fine.

**Comment by tms2003** [comment]
I used 3080x4 for training, and the results were basically normal, but the GPU was still not fully loaded, and the training time was long (640 epochs were roughly equivalent to the training of yolov8 1280). This seems to be a common problem of transformers?

In addition, I noticed that both dataloader.yml and custom_detection.yml have num_workers: 4. But in the end, custom_detection.yml seems to be invalid.

I will check my single-card environment again to see if the pytorch version is the latest.
Thanks for your work!
